packages/purpose-template-process/test/integration/getPurposeTemplateById.test.ts

```diff
@@ -0,0 +1,95 @@
+import {
+  getMockAuthData,
+  getMockContext,
+  getMockPurposeTemplate,
+  getMockValidRiskAnalysisFormTemplate,
+  sortPurposeTemplate,
+} from "pagopa-interop-commons-test";
+import {
+  generateId,
+  PurposeTemplate,
+  PurposeTemplateId,
+  purposeTemplateState,
+  TenantId,
+  tenantKind,
+} from "pagopa-interop-models";
+import { describe, expect, it } from "vitest";
+import {
+  addOnePurposeTemplate,
+  purposeTemplateService,
+} from "../integrationUtils.js";
+import {
+  purposeTemplateNotFound,
+  tenantNotAllowed,
+} from "../../src/model/domain/errors.js";
+
+describe("getPurposeTemplateById", () => {
+  it.each(Object.values(purposeTemplateState))(
+    "should get the purpose template if the requester is the creator and the purpose template is in state %s",
+    async (purposeTemplateState) => {
+      const purposeTemplate: PurposeTemplate = {
+        ...getMockPurposeTemplate(),
+        state: purposeTemplateState,
+        purposeRiskAnalysisForm: getMockValidRiskAnalysisFormTemplate(
+          tenantKind.PA
+        ),
+      };
+      await addOnePurposeTemplate(purposeTemplate);
+
+      const purposeTemplateResponse =
+        await purposeTemplateService.getPurposeTemplateById(
+          purposeTemplate.id,
+          getMockContext({
+            authData: getMockAuthData(purposeTemplate.creatorId),
+          })
+        );
+      expect({
+        ...purposeTemplateResponse,
+        data: sortPurposeTemplate(purposeTemplateResponse.data),
+      } satisfies typeof purposeTemplateResponse).toMatchObject({
+        data: sortPurposeTemplate(purposeTemplate),
+        metadata: { version: 0 },
+      });
+    }
+  );
+
+  it.each(
+    Object.values(purposeTemplateState).filter(
+      (s) => s !== purposeTemplateState.active
+    )
```

Comment:
For readability can you declare this in a const?  something like 
```suggestion
const  validPurposeTemplateState = Object.values(purposeTemplateState).filter(
      (s) => s !== purposeTemplateState.active
    )
    
    it.each(
    validPurposeTemplateState
    )
    
```

Reply:
Added here: [e638030](https://github.com/pagopa/interop-be-monorepo/pull/2304/commits/e6380307757968d2213c684f4a91680046b6ebe5)